Api stability common

IgorMinar opened the pull request.

IgorMinar committed:
docs: add api stability indicators for @angular/common

IgorMinar committed:
fix(Location): make Location#platformStrategy:LocationStrategy property private

BREAKING CHANGE: Location#platformStrategy property was previously accidentaly exported as public

If any application requires access to the current location strategy, it should be accessed via DI instead
by injecting the LocationStrategy token.

The likelyhood of anyone actually depending on this property is very low.

IgorMinar committed:
refactor(NumberPipe): remove NumberPipe and replace it with private helper function

NumberPipe was just an implementation detail that we were accidentaly exposing as a public api.

IgorMinar committed:
refactor: correct api modifier flags

IgorMinar committed:
docs: add info about __moduleName into to DirectiveMetadata docs

IgorMinar committed:
build: add missing testing.ts entry points to tsconfigs

IgorMinar committed:
test: add lots of missing stuff to the public_api_spec

- many entry points were previously missing (e.g. all testing entry points, http, etc)
- upgrade ts-api-guardian to 0.0.3 that adds support for more api surface
- add all info to the spec that was surfaced by ts-api-guardian@0.0.3

IgorMinar commented:
tests should be green now. fingers crossed.

IgorMinar committed:
style(public_api_spec): reformat

mhevery commented on modules/@angular/common/src/pipes/json_pipe.ts:
```diff
@@ -7,10 +7,10 @@ import {Json} from '../../src/facade/lang';
  * ### Example
  * {@example core/pipes/ts/json_pipe/json_pipe_example.ts region='JsonPipe'}
+ *
+ * @experimental
```
stable

mhevery commented on modules/@angular/common/src/pipes/replace_pipe.ts:
```diff
@@ -35,6 +35,8 @@ import {InvalidPipeArgumentException} from './invalid_pipe_argument_exception';
  * string properly if you are matching for regular expression special characters like parenthesis,
  * brackets etc.
+ *
+ * @experimental
```
deprecate

mhevery commented on modules/@angular/common/src/pipes/slice_pipe.ts:
```diff
@@ -58,10 +58,11 @@ import {InvalidPipeArgumentException} from './invalid_pipe_argument_exception';
  *
  * {@example core/pipes/ts/slice_pipe/slice_pipe_example.ts region='SlicePipe_string'}
+ *
+ * @stable
```
make it match `lowercase`

mhevery commented on modules/@angular/common/src/pipes/uppercase_pipe.ts:
```diff
@@ -8,9 +8,10 @@ import {InvalidPipeArgumentException} from './invalid_pipe_argument_exception';
  *
  * {@example core/pipes/ts/lowerupper_pipe/lowerupper_pipe_example.ts region='LowerUpperPipe'}
+ *
+ * @stable
```
match lowercase

mhevery commented on modules/benchpress/src/validator.ts:
```diff
@@ -7,6 +7,8 @@ import {MeasureValues} from './measure_values';
  * A valid sample is a sample that represents the population that should be observed
  * in the correct way.
+ *
+ * @experimental
```
revert

IgorMinar committed:
chore: remove old public api tests

the tests under tools/public_api_guard have better coverage - we no longer need
to maintain two copies.

IgorMinar committed:
chore: fix public_api_spec after rebase
